src/subsplit_dag.cpp

```diff
@@ -415,15 +421,16 @@ std::vector<Bitset> SubsplitDAG::GetChildSubsplits(const SizeBitsetMap &index_to
   return children_subsplits;
 }
 
-std::pair<BitsetSizeMap, SizeBitsetMap> SubsplitDAG::ProcessTopologyCounter(
-    const Node::TopologyCounter &topology_counter) {
+std::tuple<BitsetSizeMap, SizeBitsetMap, BitsetVector>
+SubsplitDAG::ProcessTopologyCounter(const Node::TopologyCounter &topology_counter) {
   BitsetSizeMap gpcsp_indexer;
   SizeBitsetMap index_to_child;
-  std::tie(rootsplits_, gpcsp_indexer, index_to_child, parent_to_range_,
+  BitsetVector rootsplits;
+  std::tie(rootsplits, gpcsp_indexer, index_to_child, parent_to_range_,
            gpcsp_count_without_fake_subsplits_) =
       SBNMaps::BuildIndexerBundle(RootedSBNMaps::RootsplitCounterOf(topology_counter),
                                   RootedSBNMaps::PCSPCounterOf(topology_counter));
-  return {gpcsp_indexer, index_to_child};
+  return {gpcsp_indexer, index_to_child, rootsplits};
 }
 
 void SubsplitDAG::CreateAndInsertNode(const Bitset &subsplit) {
```

Comment:
It's a little crazy because how big SIZE_MAX is, but I'd like to assert here that this new proposed id is less than the dag root id. 😁